### PR TITLE
BUG: Fix export .to_numpy() with nullable type

### LINKED ISSUE #40630: BUG: series.to_numpy does not work well with pd.Float64Dtype
- [ ] I have checked that this issue has not already been reported.

- [ ] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
import pandas as pd
from scipy.stats import norm

series = pd.Series([2.0, -2.5, pd.NA], dtype=pd.Float64Dtype())
norm.pdf(series)

Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "C:\Users\XXX\AppData\Local\pypoetry\Cache\virtualenvs\XXX-w19Rd76b-py3.9\lib\site-packages\scipy\stats\_distn_infrastructure.py", line 1837, in pdf
    cond1 = self._support_mask(x, *args) & (scale > 0)
  File "C:\Users\XXX\AppData\Local\pypoetry\Cache\virtualenvs\XXX-w19Rd76b-py3.9\lib\site-packages\scipy\stats\_distn_infrastructure.py", line 964, in _support_mask
    return (a <= x) & (x <= b)
  File "pandas\_libs\missing.pyx", line 360, in pandas._libs.missing.NAType.__bool__
TypeError: boolean value of NA is ambiguous
```

#### Problem description
My issue is about that pandas does not convert a pd.Series with dtype=pd.Float64Dtype() correctly to a numpy array when I try to use the scipy.stats.norm function. Therefore it cannot handle empty values (pd.NA) and the series needs to be cast right now as a float dtype to properly work which is not ideal. It probably has to do with the fact that the method series.to_numpy is called which does not work well with pd.Float64Dtype.

See also [https://github.com/scipy/scipy/issues/13729]

#### Expected Output
array([0.05399097, 0.0175283 ,        nan])

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : f2c8480af2f25efdbd803218b9d87980f416563e
python           : 3.9.1.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.19041
machine          : AMD64
processor        : Intel64 Family 6 Model 79 Stepping 1, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : English_United States.1252
pandas           : 1.2.3
numpy            : 1.20.1
pytz             : 2021.1
dateutil         : 2.8.1
pip              : 21.0.1
setuptools       : 52.0.0
Cython           : None
pytest           : 6.2.2
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.3
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : 3.0.7
pandas_gbq       : None
pyarrow          : None
pyxlsb           : None
s3fs             : None
scipy            : 1.6.1
sqlalchemy       : 1.4.2
tables           : None
tabulate         : None
xarray           : None
xlrd             : 1.2.0
xlwt             : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by taytzehao**
- [x] closes #40630
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

### CODE REVIEW

**Comment by taytzehao** [comment]
Under such a situation where a change causes cascading failed cases across multiple tests. Should I be modifying the other test cases?

**Comment by taytzehao** [comment]
If not, what strategy should I adopt?

File: pandas/core/arrays/masked.py [hunk 1]
```diff
@@ -280,10 +287,27 @@ def to_numpy(  # type: ignore[override]
...
+            # If there is NA and the data is of int type, a float
+            # is being returned as int type cannot support np.nan.
+            if is_integer_dtype(self) and self._hasna:
+                data = self._data.astype(float)
+            else:
+                data = self._data.astype(dtype)
+
+            # For numerical input, pd.na is replaced with np.nan
+            if self._hasna is True:
+                data[np.where(self._mask is True)] = np.nan
```
Comment (mzeitlin11) [inline pandas/core/arrays/masked.py:308]: This line looks suspect ... `self._mask` is an ndarray, so won't comparing identity with True just always give False?

Comment (mzeitlin11) [reply pandas/core/arrays/masked.py:308]: Also, what about the specified na_value?

File: pandas/core/arrays/masked.py [hunk 2]
```diff
@@ -280,10 +287,27 @@ def to_numpy(  # type: ignore[override]
...
+                # error: Incompatible types in assignment (expression has type
+                # "Type[object]", variable has type "Union[str, dtype[Any], None]")
+                dtype = object  # type: ignore[assignment]
+
+        if is_numeric_dtype(self):
+
+            # If there is NA and the data is of int type, a float
+            # is being returned as int type cannot support np.nan.
+            if is_integer_dtype(self) and self._hasna:
+                data = self._data.astype(float)
```
Comment (mzeitlin11) [inline pandas/core/arrays/masked.py:302]: Doesn't this ignore any specified dtype?

**Review by mzeitlin11 — COMMENTED** [review]
Thanks for the pr @taytzehao. Some comments, plus not 100% sure this is what should be done anyway. 

The whole purpose of the new missing value `NA` is that it means something different than `np.nan`. Because of that, defaulting to object type makes sense to me because that is the common type between int or float and `NA`. Also, casting ints to float is not lossless.  The behavior of replacing `NA` with `np.nan` can still be achieved by specifying a `na_value`.

**Comment by mzeitlin11** [comment]
> Under such a situation where a change causes cascading failed cases across multiple tests. Should I be modifying the other test cases?

There are a bunch of test failures, so I think the approach should be looked at more carefully first. Running tests locally should also always be the first step to catch test failures before using CI

File: pandas/tests/arrays/test_numpy.py [hunk 3]
```diff
@@ -154,6 +154,21 @@ def test_to_numpy():
     tm.assert_numpy_array_equal(result, expected)
 
 
+@pytest.mark.parametrize(
+    "data_content,data_type,expected_result",
+    [
+        ([1, 2, 3], pd.Float64Dtype(), np.array([1, 2, 3], dtype=np.float64)),
```
Comment (datapythonista) [inline pandas/tests/arrays/test_numpy.py:160]: Why not build the Series here instead? I think just two parameters `data` and `expected_results` is enough (I think `expected` is descriptive enough and more common, but no big deal).

File: doc/source/whatsnew/v1.3.0.rst [hunk 4]
```diff
@@ -708,6 +708,7 @@ Conversion
 - Bug in :func:`factorize` where, when given an array with a numeric numpy dtype lower than int64, uint64 and float64, the unique values did not keep their original dtype (:issue:`41132`)
 - Bug in :class:`DataFrame` construction with a dictionary containing an arraylike with ``ExtensionDtype`` and ``copy=True`` failing to make a copy (:issue:`38939`)
 - Bug in :meth:`qcut` raising error when taking ``Float64DType`` as input (:issue:`40730`)
+- Bug in :meth:`BaseMaskedArray.to_numpy` does not output ``numeric_dtype`` with ``numeric_dtype`` input (:issue:`40630`)
```
Comment (datapythonista) [inline doc/source/whatsnew/v1.3.0.rst:711]: I find this sentence a bit difficult to understand, can you rephrase please?

**Review by datapythonista — COMMENTED** [review]
As mentioned, your approach doesn't seem to be working, you'll have to review. In general, what you'd want to do to fix a bug is:
- Implement a test, as you did, and run the original code and make sure the test fails
- Then change the code, and make sure the both the new test and all existing tests work

So, answering your question, you shouldn't modify the tests that are broken with your implementation (except in very rare cases where the tests were wrong, which is not the case).
